packages/react-components/react-button/src/components/Button/useButton.ts

```diff
@@ -56,7 +55,7 @@ export const useButton_unstable = (
           type: 'button', // This is added because the default for type is 'submit'
         },
       }),
-    ),
+    ) as ButtonState['root'],
```

Comment:
Because we now have proper getNativeProps types, elements that can have variadic root cannot match with original `Props` definition where `Slot` mapped type is used to define the api.

Why?  

`Slot` api design is the culprit that cannot match with implementation ever - because we cannot extract first type from union nor transform unions to arrays.